File: app/src/main/java/com/shindra/stationActivity.java
package com.shindra;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.shindra.arrakis.observable.ObservableExtensionKt;
import com.shindra.arrakis.observable.ObservableListener;
import com.shindra.ctslibrary.apibo.RouteType;
import com.shindra.ctslibrary.bo.Line;

import org.jetbrains.annotations.NotNull;

public class stationActivity extends AppCompatActivity {

    private stationFragment frag;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_station_diary);

        getSupportActionBar().setTitle(getString(R.string.stationActivityName));

        frag = new stationFragment();
        Bundle bundle = new Bundle();

        getSupportFragmentManager().beginTransaction().add(R.id.container_station,frag).commit();
        frag.setArguments(bundle);
        bundle.putString("lineName",getIntent().getStringExtra("TRAM_LINE"));


    }

}

File: app/src/main/java/com/shindra/stationActivity.kt
package com.shindra

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class stationActivity : AppCompatActivity() {
    private var frag: stationFragment? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_station_diary)
        supportActionBar!!.title = getString(R.string.stationActivityName)

        //Create a new fragment stationFragment and his bundle
        frag = stationFragment()
        val bundle = Bundle()

        //begin transaction between the container and the fragment frag. Add parameter to the bundle
        supportFragmentManager.beginTransaction().add(R.id.container_station, frag!!).commit()
        bundle.putString("lineName", intent.getStringExtra("TRAM_LINE"))
        frag!!.arguments = bundle
    }
}